modules/configuration/src/main/java/org/apache/ignite/internal/configuration/ConfigurationChanger.java

```diff
@@ -692,49 +705,55 @@ private void validateConfiguration(SuperRoot curRoots, SuperRoot changes) {
-            SuperRoot newSuperNoDefaults = oldSuperRootNoDefaults.copy();
+                SuperRoot oldSuperRoot = oldStorageRoots.roots;
+                SuperRoot oldSuperRootNoDefaults = oldStorageRoots.rootsWithoutDefaults;
+                SuperRoot newSuperRoot = oldSuperRoot.copy();
+                SuperRoot newSuperNoDefaults = oldSuperRootNoDefaults.copy();
 
-            Map<String, ?> dataValuesPrefixMap = toPrefixMap(changedValues);
+                Map<String, ?> dataValuesPrefixMap = toPrefixMap(changedValues);
 
-            compressDeletedEntries(dataValuesPrefixMap);
+                compressDeletedEntries(dataValuesPrefixMap);
 
-            fillFromPrefixMap(newSuperRoot, dataValuesPrefixMap);
-            fillFromPrefixMap(newSuperNoDefaults, dataValuesPrefixMap);
+                fillFromPrefixMap(newSuperRoot, dataValuesPrefixMap);
+                fillFromPrefixMap(newSuperNoDefaults, dataValuesPrefixMap);
 
-            long newChangeId = changedEntries.changeId();
+                long newChangeId = changedEntries.changeId();
 
-            var newStorageRoots = new StorageRoots(newSuperNoDefaults, newSuperRoot, mergeData(oldStorageRoots.data, changedEntries));
+                var newStorageRoots = new StorageRoots(newSuperNoDefaults, newSuperRoot, mergeData(oldStorageRoots.data, changedEntries));
 
-            rwLock.writeLock().lock();
+                rwLock.writeLock().lock();
 
-            try {
-                storageRoots = newStorageRoots;
-            } finally {
-                rwLock.writeLock().unlock();
-            }
+                try {
+                    storageRoots = newStorageRoots;
+                } finally {
+                    rwLock.writeLock().unlock();
+                }
 
-            long notificationNumber = notificationListenerCnt.incrementAndGet();
+                long notificationNumber = notificationListenerCnt.incrementAndGet();
 
-            CompletableFuture<Void> notificationFuture = configurationUpdateListener
-                    .onConfigurationUpdated(oldSuperRoot, newSuperRoot, newChangeId, notificationNumber);
+                CompletableFuture<Void> notificationFuture = configurationUpdateListener
+                        .onConfigurationUpdated(oldSuperRoot, newSuperRoot, newChangeId, notificationNumber);
 
-            return notificationFuture.whenComplete((v, t) -> {
-                if (t == null) {
-                    oldStorageRoots.changeFuture.complete(null);
-                } else {
-                    oldStorageRoots.changeFuture.completeExceptionally(t);
-                }
-            });
+                return notificationFuture.whenComplete((v, t) -> {
+                    if (t == null) {
+                        oldStorageRoots.changeFuture.complete(null);
+                    } else {
+                        oldStorageRoots.changeFuture.completeExceptionally(t);
+                    }
+                });
+            } catch (Throwable e) {
```

Comment:
Note to reviewer: without this "catch", errors happening outside of `notificationFuture` will never be propagated to the caller.